include/envoy/http/protocol.h

```diff
@@ -10,5 +11,24 @@ namespace Http {
 enum class Protocol { Http10, Http11, Http2 };
 const size_t NumProtocols = 3;
 
+static const std::string DefaultString = "";
+static const std::string Http10String = "HTTP/1.0";
```

Comment:
Static non-POD is not allowed in the Envoy style, see https://github.com/envoyproxy/envoy/blob/master/STYLE.md. This could probably just be `const char DefaultString = ""` etc.

Reply:
We could also put these in a struct in headers.h
I was wavering on asking that anyway - they're not so much headers as "part of the firstline" but it's where I'd look for predefined header-type strings.

Reply:
IMO we should continue to return string references for perf reasons. I still think that the non-POD rule for std::string is a little silly, but, the rules are the rules. I would just use the constant class pattern we use elsewhere like in Http::Headers.

Reply:
@mattklein123 i think you are referring to https://github.com/envoyproxy/envoy/blob/master/source/common/http/headers.h
if so then i can certainly adopt that patter here as well.
@alyssawilk is it ok if i keep the new constant class here?

Reply:
Either way is fine with me

Reply:
@mattklein123 @alyssawilk Last push addresses this. Adding `NOT_REACHED` in this file causes a circular dependency so i added a `abort()`. If that is not what we want then I can move both constants and the `getProtocolString` to the `common/http/headers.h` file.
please let me know if that would be preferable.

Reply:
I would recommend moving into common and using NOT_REACHED. You can either put in headers.h or feel free to just put in Http::Utility

Reply:
cool. moved. please review when u folks get a chance. thanks